discovery_engine_core/web-api2/src/middleware/tracing.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use futures_util::FutureExt;
+use serde::Serialize;
+use std::future::Future;
+
+use actix_web::{
+    dev::{Service, ServiceRequest, ServiceResponse},
+    HttpMessage,
+};
+
+use tracing::{trace, Instrument};
+use tracing_actix_web::root_span_macro::private::tracing::info_span;
+use uuid::Uuid;
+
+#[derive(Clone, Copy, Debug, derive_more::Display, Serialize)]
+#[serde(transparent)]
+pub(crate) struct RequestId(Uuid);
+
+impl RequestId {
+    pub(crate) fn generate() -> Self {
+        Self(Uuid::new_v4())
+    }
+
+    pub(crate) fn missing() -> Self {
+        Self(Uuid::nil())
+    }
+}
+
+pub(crate) fn tracing_log_request<S, B>(
+    request: ServiceRequest,
+    service: &S,
+) -> impl Future<Output = Result<ServiceResponse<B>, actix_web::Error>> + 'static
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = actix_web::Error>,
+    S::Future: 'static,
+{
+    //FIXME try to use tracing-actix-web and the request id of it instead
```

Comment:
is this fix planned for this PR?

Reply:
so maybe let's remove a `FIXME` comment and make sure that we handle the `tracing-actix-web` in opentelemetry integration story https://xainag.atlassian.net/browse/ET-2952